src/Symfony/Component/Validator/Constraints/GroupSequence.php

```diff
@@ -56,7 +56,7 @@ class GroupSequence
     /**
      * The groups in the sequence.
      *
-     * @var string[]|string[][]|GroupSequence[]
+     * @var array<string|string[]|GroupSequence>
```

Comment:
Do we actually have tests that confirm that nested group sequences did ever fully work?

Reply:
We don't, because constraints are empty on the only test case that does so... But a deep TypeError looks like something is like something is missing. Maybe just a check with a clear error message. The doc mentions nested groups btw.

Reply:
Nested groups should be okay. The `GroupSequence` part looks suspicious to me. What error would we get when changing the annotation to `array<string|string[]>|GroupSequence`?

Reply:
[`RecursiveContextualValidator`](https://github.com/symfony/symfony/blob/479919d4d5cbc17710d16a04ff9728d1519afb8e/src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php#L581-L582) has many `instanceof GroupSequence` checks that map to this possibility. But `FormValidator` doesn't handle it at all.